backend/services/implementations/classService.ts

```diff
@@ -106,10 +108,18 @@ class ClassService implements IClassService {
 
   async getClassesByTeacherId(
     teacherId: string,
+    queryOptions?: ClassQueryOptions,
   ): Promise<Array<ClassResponseDTO>> {
     let classes: Class[];
     try {
-      classes = await MgClass.find({ teacher: { $eq: teacherId } });
+      const query = MgClass.find({
+        teacher: { $eq: teacherId },
+      });
+      applyQueryOptions(query, queryOptions);
+      if (queryOptions?.excludeArchived) {
+        query.where({ isActive: { $in: [true, undefined] } });
```

Comment:
`undefined` wouldn't be needed if our data was correct right? cause `isActive` is default set to `true` on class creation

Reply:
Yes, but right now our database contains a lot of classes where this field is unset. I figured it was probably better to just add the check than do a big migration

Reply:
hmm, ig we prob wouldn't want that check in prod tho? you can add it for testing purposes or at least put TODO to remove

Reply:
idk up to you, maybe better for backwards compatibility purposes

Reply:
yeah, my thinking is basically that it doesn't hurt to have this check in prod, and it makes the code overall safer, so unless it starts causing a performance problem I think it's best to keep it